### PR TITLE
Remove invalid column_view usage in string-scalar-to-column function

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes the `make_column_from_scalar` function for `string_scalar` internal usage of a temporary `column_view` with non-zero size but no data or children to call `cudf::strings::detail::fill`. This relied too much on fragile internal logic which has cause several headaches including the recent work adding prefetch logic to libcudf.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/column/column_factories.cu [hunk 1]
```diff
@@ -57,15 +58,28 @@ std::unique_ptr<cudf::column> column_from_scalar_dispatch::operator()<cudf::stri
...
-  return output;
+  // fill the column with the scalar
+  rmm::device_uvector<cudf::strings::detail::string_index_pair> indices(size, stream);
+  thrust::generate(
+    rmm::exec_policy_nosync(stream), indices.begin(), indices.end(), [d_str] __device__() {
+      // special handling for empty string -- from an all-empty strings column
+      return d_str.empty()
+               ? cudf::strings::detail::string_index_pair{"", 0}
+               : cudf::strings::detail::string_index_pair{d_str.data(), d_str.size_bytes()};
+    });
```
Comment (ttnghia) [inline cpp/src/column/column_factories.cu:81]: Should `thrust::uninitialized_fill` be better for this case?
```suggestion
  auto const row_pair = d_str.empty()
               ? cudf::strings::detail::string_index_pair{"", 0}
               : cudf::strings::detail::string_index_pair{d_str.data(), d_str.size_bytes()};
  thrust::uninitialized_fill(rmm::exec_policy_nosync(stream), indices.begin(), indices.end(), row_pair);
```

**Review by ttnghia — APPROVED** [review]

**Review by vyasr — COMMENTED** [review]
Trying to understand this. Are the checks for an invalid or empty value sufficient to make calling `strings::detail::fill` safe, or is it still fragile even in that scenario? If it is safe, are you choosing the generate->make_strings_column approach so that you can use the same code path in all cases?

**Comment by davidwendt** [comment]
> Trying to understand this. Are the checks for an invalid or empty value sufficient to make calling `strings::detail::fill` safe, or is it still fragile even in that scenario? If it is safe, are you choosing the generate->make_strings_column approach so that you can use the same code path in all cases?

It is not safe to call `fill` since `fill` expects a column of rows and I was using a trick where I created a strings `column_view` with non-zero size and a null-mask but no children (or data). This is fragile and now removed by performing the scalar-to-column logic without using `fill`. The code here is simpler (and probably faster) than the `fill` logic since every row is created with the same value.

File: cpp/src/column/column_factories.cu [hunk 2]
```diff
@@ -20,10 +20,11 @@
 #include <cudf/dictionary/dictionary_factories.hpp>
 #include <cudf/lists/detail/lists_column_factories.hpp>
 #include <cudf/scalar/scalar.hpp>
-#include <cudf/strings/detail/fill.hpp>
+#include <cudf/strings/detail/strings_column_factories.cuh>
 
 #include <rmm/resource_ref.hpp>
 
+#include <thrust/generate.h>
```
Comment (ttnghia) [inline cpp/src/column/column_factories.cu:27]:
```suggestion
#include <thrust/uninitialized_fill.h>
```

**Review by vyasr — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Pull request merged**